### DISCUSSION TITLE
Akka.Persistence.Sql.Query GetReadJournal

### DISCUSSION THREAD

**Discussion by woodpk** [akkadotnet/akka.net#6789] [Q&A]
Hello Community.
I'm needing some help with what I assume is pretty basic functioning for this package. I am unable to get a read journal for the new Akka.Persistence.Sql.Query SqlReadJournal.  
Using the GetReadJournal method from the provided SqlReadJournalProvider, I am getting the exception: Akka.Configuration.ConfigurationException: The configuration property akka.persistence.journal.sql.table-mapping is null or empty.
This is somewhat confusing because when looking at the config in debug mode, the table-mapping property is set to default, but when the process attempts to actually run through config.GetString("table-mapping") in the JournalTableConfig class, it returns a null value.

When I manually do a Config.GetString, but use the full path ("akka.persistence.journal.sql.table-mapping"), it correctly returns the 'default' value, but the JournalTableConfig class is using 'table-mapping' as its search string, which is returning null when parsing the entire config hocon tree.

As a result I am not able to instantiate a read journal.
For writes, it seems this package is working fine as I am able to write to a test database with the default table configuration.

So, I know others have been able to get persistence query to work and I would love to know how. Thanks for your help and attention.

Stacktrace is:
Akka.Configuration.ConfigurationException: The configuration property akka.persistence.journal.sql.table-mapping is null or empty
   at Akka.Persistence.Sql.Config.JournalTableConfig..ctor(Config config)
   at Akka.Persistence.Sql.Config.ReadJournalConfig..ctor(Config config)
   at Akka.Persistence.Sql.Query.SqlReadJournal..ctor(ExtendedActorSystem system, Config config, String configPath)
   at Akka.Persistence.Sql.Query.SqlReadJournalProvider.GetReadJournal()
   at PRCS.WorkItems.Tests.TestConfig.<>c__DisplayClass4_0.<ConfigSqlServer>b__0(ActorSystem system, IActorRegistry registry) in D:\PRCS.WorkItems\PRCS.WorkItems.Tests\Configuration.cs:line 89
   at Akka.Hosting.AkkaConfigurationBuilder.<>c__DisplayClass25_0.<ToAsyncStarter>g__Starter|0(ActorSystem f, IActorRegistry registry)
   at Akka.Hosting.AkkaConfigurationBuilder.StartAsync(ActorSystem sys)

**Comment by Aaronontheweb** [comment]
What does your HOCON / Akka.Hosting look like @woodpk ?

**Reply by woodpk** [reply]
@Aaronontheweb, here you go:

In this instance, I am using the default config.  I am trying to create the read journal in config because I was initially testing a persistent view actor w/ the ireadjournal as a constructor prop.
I did testing using 2 different namespaces.  I figured using the existing akka.persistence.query wouldn't work, but don't understand why using akka.persistence.sql.query 's own journal provider didn't get me the journal.  I am using the variable 'prop' just to see if Config's GetString method will get the "default" value of the hocon when i give it the full path.

        builder
            .WithSqlPersistence(
				connectionString: sqlStringBuilderConfig.ConnectionString,
	            providerName: LinqToDB.ProviderName.SqlServer2022
            )
            //.AddHocon(hocon, HoconAddMode.Append)
            .StartActors((system, registry) =>
            {
	            //Akka.Persistence.Query
					//var readJournal = system.ReadJournalFor<SqlReadJournal>(SqlReadJournal.Identifier);
	            
	            
	            //Akka.Persistence.Sql.Query
		            //testing getting the string w/ full path
		            var prop = Akka.Persistence.Sql.SqlPersistence.DefaultConfiguration.GetString("akka.persistence.journal.sql.table-mapping");
		            
		            
		            var journalProvider = new SqlReadJournalProvider(
			            system.AsInstanceOf<ExtendedActorSystem>()
			            , Akka.Persistence.Sql.SqlPersistence.DefaultConfiguration);

		            var journal = journalProvider.GetReadJournal();


	            CreateActors(system, registry, actorIds); //, readJournal);
            });

**Reply by Aaronontheweb** [reply]
Thanks @woodpk - I've pinged @Arkatufus (dev who owns this) in our chat but he's going to be out of the office for a few days. He'll get back to you when he can.

**Reply by woodpk** [reply]
Thanks you @Aaronontheweb . I look forward to his expertise.

**Comment by Arkatufus** [comment]
This looks like a bug to me, I'll work on it as soon as I can

**Reply by Aaronontheweb** [reply]
@Arkatufus this has been fixed no?

**Comment by woodpk** [comment]
Thanks for looking into the possible bug. I was wondering if you had found anything?


-Patrick

________________________________

**Reply by Arkatufus** [reply]
I think it is already fixed in https://github.com/akkadotnet/Akka.Persistence.Sql/pull/283, are you still having problem?
